src/org/python/core/PySystemState.java

```diff
@@ -1022,16 +1022,27 @@ private static void initRegistry(Properties preProperties, Properties postProper
      */
     private static String getConsoleEncoding(Properties props) {
 
-        // From Java 8 onwards, the answer may already be to hand in the registry:
-        String encoding = props.getProperty("sun.stdout.encoding");
-        String os = props.getProperty("os.name");
+        // Java 19+
+        String encoding = props.getProperty("stdout.encoding");
+        if (encoding != null) {
+            // Windows: cp65001 is automatically mapped to UTF-8, no additional processing is needed
+            return encoding;
+        }
 
+        // Java 8 to 18
+        encoding = props.getProperty("sun.stdout.encoding");
         if (encoding != null) {
+            // Windows: these versions of Java return "cp65001" for UTF-8
+            if (encoding.equals("cp65001")) {
```

Comment:
Shouldn't we only bother with this inside the "if it's Windows" clause?

Reply:
Yes, good catch. Fixed.